### PR TITLE
Locations page

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://ethyca.atlassian.net/browse/PROD-1491

### Description Of Changes

Adds a locations page! 

https://github.com/ethyca/fides/assets/24641006/9df012ac-7659-4ccd-8560-91c4413b821f

### Code Changes

* [x] New page for locations
* [x] `PickerCard` to be reused by regulations page
* [x] Checkbox behavior and grouping by continents
* [x] Search bar + "regulated" toggle
* [x] Cypress tests

### Steps to Confirm

The preview env won't work here because the backend changes from https://github.com/ethyca/fidesplus/pull/1269 are needed

* [ ] Run fidesplus on this branch: https://github.com/ethyca/fidesplus/pull/1269
* [ ] Run the admin UI via `turbo run dev` in the admin UI. the feature flag is only enabled for dev/test environments
* [ ] Click the settings gear icon
* [ ] You should see a new "Locations" option in the side nav. Click it
* [ ] Check and uncheck various locations and make sure all works as expected

Note: the "View more" button is a separate ticket: https://ethyca.atlassian.net/browse/PROD-1493

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`

### CODE REVIEW

**Comment by allisonking** [comment]
Current status


https://github.com/ethyca/fides/assets/24641006/615f96ac-a4a7-472c-a449-fff6b9729c00

File: clients/admin-ui/src/features/locations/transformations.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  const allContinents = new Set(locations.map((l) => l.continent).sort());
+  allContinents.forEach((continent) => {
+    byContinent[continent] = locations
+      .filter((l) => l.continent === continent)
+      .sort((a, b) => a.name.localeCompare(b.name));
+  });
+  return byContinent;
+};
+
+export const groupByBelongsTo = (
```
Comment (allisonking) [inline clients/admin-ui/src/features/locations/transformations.ts:16]: not used yet, but I think will be by https://ethyca.atlassian.net/browse/PROD-1493

**Review by TheAndrewJackson — APPROVED** [review]
UAT: tested locally and working :+1: 

@allisonking This works great! Why wasn't formik used for form? Not a blocker but I'm curious what it was done this way.

**Comment by allisonking** [comment]
> UAT: tested locally and working 👍
> 
> @allisonking This works great! Why wasn't formik used for form? Not a blocker but I'm curious what it was done this way.

heh good question! I did it this way first, and then thought "wait maybe I should've used formik here" and then spent some time refactoring it to use formik but the logic actually became quite complicated with using nested array keys (so the name of a checkbox might be `North America[2].selected` for "California" and that was difficult to grok). in the end I reverted back to this method since formik is great for anything that needs validation or required fields (most forms!) but for a list of checkboxes, it was causing me more confusion than it was helping me. I did save my half done refactor to another branch just in case though...

**Pull request merged**